@here/harp-mapview/lib/PickHandler.ts

```diff
@@ -232,6 +247,46 @@ export class PickHandler {
         return pickResults;
     }
 
+    private intersectMapObjects2D(x: number, y: number, intersects: THREE.Intersection[]) {
+        // Find 2D objects at the picking ray intersection on the ground.
+        const rayCaster = this.rayCaster;
+        rayCaster.setFromCamera(
+            this.mapView.getNormalizedScreenCoordinates(x, y),
+            this.mapView.camera
+        );
+        const worldGroundIntersect = rayCaster.intersectGround(x, y, true);
+        if (!worldGroundIntersect) {
+            return;
+        }
+        const geoGroundIntersect = this.mapView.projection.unprojectPoint(worldGroundIntersect);
+
+        // const tileOffset = 0; // TODO: FIND TILE OFFSET!
+        // const intersectTiles = this.mapView.visibleTileSet.getRenderedTilesAtLocation(
+        //     geoGroundIntersect,
+        //     tileOffset
+        // );
+        // if (intersectTiles.length === 0) {
+        //     return;
+        // }
+
+        // const intersectNormal = new THREE.Vector3();
+        // this.mapView.projection.surfaceNormal(worldGroundIntersect, intersectNormal);
+        // rayCaster.set(worldGroundIntersect.sub(rayCaster.camera.position), intersectNormal);
+        rayCaster.camera = this.camera;
+        rayCaster.set(
+            this.camera.position,
+            worldGroundIntersect.sub(this.mapView.camera.position).normalize()
+        );
+
+        // for (const tile of intersectTiles) {
+        //     rayCaster.intersectObjects(tile.objects, true, intersects);
+        // }
+
+        for (const object of this.mapView.worldRootObject.children) {
```

Comment:
do you need the loop here? If you pass recursive to intersectObject it should do it internally ?